packages/shared-kernel/src/crypto/__tests__/Amount.spec.ts

```diff
@@ -44,6 +61,23 @@ describe('Given the Amount type', () => {
     });
   });
 
+  describe('when dividing an Amount by another', () => {
+    it('should return the expected Amount', () => {
+      const amount1 = Amount.usd(42);
+      const amount2 = Amount.usd(2);
+
+      const expected = Amount.usd(21);
+      expect(amount1.div(amount2)).toEqual(expected);
```

Comment:
Would division (like multiplication) involve a scalar value rather than another amount?

In other words:
```
amount1.div(2)
```

In that case the invariant check is not even needed.

The module defines an `AmountValue` that is the scalar type we could use (see `mul` method for example).

Reply:
I thought about it, and it made a lot of sense to support scalar division, so I changed the method to support it.